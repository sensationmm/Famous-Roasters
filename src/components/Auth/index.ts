import { Authenticator } from './Authenticator'
import { AuthFormEmail } from './AuthFormEmail'
import { AuthFormItemInput } from './AuthFormItemInput'
import { AuthFormPassword } from './AuthFormPassword'
import { AuthSignIn } from './SignIn'

export { Authenticator as Authenticator }
export { AuthFormEmail as AuthFormEmail }
export { AuthFormItemInput as AuthFormItemInput }
export { AuthFormPassword as AuthFormPassword }
export { AuthSignIn as AuthSignIn }
