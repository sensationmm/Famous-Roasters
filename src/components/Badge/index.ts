import { Badge } from './Badge'

export { Badge as Badge }
