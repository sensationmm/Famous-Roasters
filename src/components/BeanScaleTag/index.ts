import { BeanScaleTag } from './BeanScaleTag'

export { BeanScaleTag as BeanScaleTag }
