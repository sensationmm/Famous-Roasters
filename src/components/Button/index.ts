import { Button, Color, Emphasis, Size } from './Button'

export { Button as Button }
export { Color as ButtonColor }
export { Emphasis as ButtonEmphasis }
export { Size as ButtonSize }
