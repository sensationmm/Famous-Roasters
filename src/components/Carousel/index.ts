import { Carousel } from './Carousel'

export { Carousel as Carousel }
