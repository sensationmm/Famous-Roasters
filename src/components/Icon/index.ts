import { Icon, IconName, IconSize } from './Icon'

export { Icon as Icon }
export { IconName as IconName }
export { IconSize as IconSize }
