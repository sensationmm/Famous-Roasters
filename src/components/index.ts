import { Button, ButtonColor, ButtonEmphasis, ButtonSize } from './Button'
import { Footer } from './Footer'
import { Layout } from './Layout'
import { Navigation } from './Navigation'
import { Typography, TypographySize, TypographyType } from './Typography'

export { Button as Button }
export { ButtonColor as ButtonColor }
export { ButtonEmphasis as ButtonEmphasis }
export { ButtonSize as ButtonSize }
export { Footer as Footer }
export { Layout as Layout }
export { Navigation as Navigation }
export { Typography as Typography }
export { TypographyType as TypographyType }
export { TypographySize as TypographySize }
