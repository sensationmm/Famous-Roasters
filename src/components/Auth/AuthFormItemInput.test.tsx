import { MockedProvider } from '@apollo/client/testing'
import { fireEvent, render, screen, waitFor } from '@testing-library/react'
import Form from 'rc-field-form'
import React from 'react'
import { act } from 'react-dom/test-utils'
import { Icon, IconName } from 'src/components/Icon'

import { AuthFormItemInput } from '.'

describe('ConfirmSignUp custom auth component', () => {
  const snippet = (type?: string) => (
    <Form name="testAuthFormItemInput" method="POST">
      <AuthFormItemInput name="itemName" label="itemLabel" type={type} rules={[]} placeholder="itemPlaceHolder" />
    </Form>
  )

  it('Renders correctly', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet('password')}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly for non given type', async () => {
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {snippet(undefined)}
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly with errors', async () => {
    // needed as async validator throws a console warning on input too short
    jest.spyOn(console, 'warn').mockImplementation(jest.fn())
    const { container } = render(
      <MockedProvider defaultOptions={{ watchQuery: { fetchPolicy: 'no-cache' } }}>
        {
          <Form name="testAuthFormItemInput" method="POST">
            <AuthFormItemInput
              dataTestId="input-field"
              name="itemName"
              label="itemLabel"
              type={'password'}
              rules={[{ min: 5, message: 'Not long enough' }]}
              placeholder="itemPlaceHolder"
              validateTrigger="onChange"
              icon={<Icon name={IconName.PasswordShow} />}
            />
          </Form>
        }
      </MockedProvider>,
    )
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    const input = await screen.findByTestId('input-field')
    expect(input).toBeInTheDocument()
    await act(() => {
      fireEvent.change(input, { target: { value: 'asd' } })
    })
    expect(container).toMatchSnapshot()
    jest.spyOn(console, 'warn').mockRestore()
  })
})
