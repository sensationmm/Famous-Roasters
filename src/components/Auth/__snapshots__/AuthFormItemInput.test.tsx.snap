// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmSignUp custom auth component Renders correctly 1`] = `
<div>
  <form
    method="POST"
  >
    <div
      class="relative"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            itemLabel
          </span>
        </span>
        <input
          class="w-full  border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          placeholder="itemPlaceHolder"
          type="password"
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`ConfirmSignUp custom auth component Renders correctly for non given type 1`] = `
<div>
  <form
    method="POST"
  >
    <div
      class="relative"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            itemLabel
          </span>
        </span>
        <input
          class="w-full  border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          placeholder="itemPlaceHolder"
          type="text"
          value=""
        />
      </label>
    </div>
  </form>
</div>
`;

exports[`ConfirmSignUp custom auth component Renders correctly with errors 1`] = `
<div>
  <form
    method="POST"
  >
    <div
      class="relative"
    >
      <label
        class="relative"
      >
        <span
          class="absolute -top-6 left-3 px-1 bg-white"
        >
          <span
            class="text-sm leading-5"
          >
            itemLabel
          </span>
        </span>
        <input
          class="w-full border-negative pr-11 border rounded-full focus:outline-0 focus:ring px-4 py-2 focus:ring-primary/30"
          data-testid="input-field"
          placeholder="itemPlaceHolder"
          type="password"
          value="asd"
        />
      </label>
      <div
        class="absolute right-3 top-1/2 -translate-y-2/4"
      >
        <svg
          class="undefined"
          fill="inherit"
          height="24"
          viewBox="0 0 22 16"
          width="24"
        >
          <path
            clip-rule="evenodd"
            d="M14 8C14 9.65685 12.6569 11 11 11C9.34315 11 8 9.65685 8 8C8 6.34315 9.34315 5 11 5C12.6569 5 14 6.34315 14 8Z M1.45825 7.99997C2.73253 3.94288 6.52281 1 11.0004 1C15.4781 1 19.2684 3.94291 20.5426 8.00004C19.2684 12.0571 15.4781 15 11.0005 15C6.52281 15 2.73251 12.0571 1.45825 7.99997Z"
            fill="none"
            fill-rule="evenodd"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
    <ul
      class="text-negative mt-2 mb-4"
    >
      <li
        class="text-xs leading-4"
      >
        Not long enough
      </li>
    </ul>
  </form>
</div>
`;
